Reworking of includes and layouts episodes as per #208

anenadic opened the pull request:
Reworking of includes and layouts episodes:

- code reuse examples now use top navigation links instead of banner and footer
- layouts inheritance is shown on the example of home layout inheriting from default layout and adding a banner image

Example code repository for reference: https://github.com/anenadic/group-website

anenadic committed:
Reworking of includes and layouts episodes as per #208

anenadic committed:
Fixed navigation links in example code

anenadic committed:
Typo fixes

anenadic committed:
Typo fix + a point about reusing existing themes

tobyhodges commented on _episodes/includes.md:
```diff
@@ -81,131 +81,104 @@ the sites linked in the previous exercise aren't built with Jekyll.
-why don't we try adding a banner to the top of each page?
+Websites typically have some navigation links at the top of each page to help visitors navigate to various portions of the site.
+To make these links appear above every page on our site,
+we could add the same code above to each Markdown file in our repository.
```
```suggestion
we could add the same code immediately after the YAML header of each Markdown file in our repository.
```

tobyhodges commented on _episodes/includes.md:
```diff
@@ -81,131 +81,104 @@ the sites linked in the previous exercise aren't built with Jekyll.
+Websites typically have some navigation links at the top of each page to help visitors navigate to various portions of the site.
+To make these links appear above every page on our site,
+we could add the same code above to each Markdown file in our repository.
+But if we wanted to adjust the - adjust the navigation link target, add a new navigation link, remove a link, etc. -
```
```suggestion
But if we wanted to adjust the menu - adjust the navigation link target, add a new navigation link, remove a link, etc. -
```

tobyhodges commented on _episodes/includes.md:
```diff
@@ -81,131 +81,104 @@ the sites linked in the previous exercise aren't built with Jekyll.
+To make these links appear above every page on our site,
+we could add the same code above to each Markdown file in our repository.
+But if we wanted to adjust the - adjust the navigation link target, add a new navigation link, remove a link, etc. -
+we would need to make the same adjustment on every page.
```
```suggestion
we would need to make the same adjustment on every page.
This is both time-consuming and error-prone: 
it would be easy to accidentally mistype a link or forget to update one of the files.
```

tobyhodges commented on _episodes/includes.md:
```diff
@@ -81,131 +81,104 @@ the sites linked in the previous exercise aren't built with Jekyll.
+    <hr>{% endraw %}
+    ~~~
+The snippet will create a table with a single row of links followed by a horizontal line separator. 
+5. Now insert the following `_include` directive at the beginning of `index.md`:
```
```suggestion
5. Now insert the following `include` directive at the beginning of `index.md`:
```

tobyhodges commented on _episodes/includes.md:
```diff
@@ -81,131 +81,104 @@ the sites linked in the previous exercise aren't built with Jekyll.
+we would need to make the same adjustment on every page.
+Instead, we can go some way to avoid this hassle by using
+some magic that Jekyll provides: `include` tags.
+To demonstrate this, we will save the HTML snippet for creating navigation links into
```
```suggestion
To demonstrate this, we will save an HTML snippet for creating navigation links into
```

tobyhodges commented on _episodes/includes.md:
```diff
@@ -81,131 +81,104 @@ the sites linked in the previous exercise aren't built with Jekyll.
+But if we wanted to adjust the - adjust the navigation link target, add a new navigation link, remove a link, etc. -
+we would need to make the same adjustment on every page.
+Instead, we can go some way to avoid this hassle by using
+some magic that Jekyll provides: `include` tags.
```
```suggestion
some magic that Jekyll provides: `include` tags.

```

I think a paragraph break helps readability here.

tobyhodges commented on _episodes/includes.md:
```diff
@@ -81,131 +81,104 @@ the sites linked in the previous exercise aren't built with Jekyll.
+             
+    {% include navigation.html %}
+             
+    # {{ site.title }}
```
```suggestion
    # Building Websites in GitHub
```

`{{ site.title }}` was removed in #241

tobyhodges commented on _episodes/includes.md:
```diff
@@ -224,10 +197,12 @@ We will see another example of this shortly.
 {: .callout }
 
-> ## Exercise: Including Contact Information
+## Reusing Footer 
```
```suggestion
## Reusable Footer 
```

tobyhodges commented on _episodes/includes.md:
```diff
@@ -224,10 +197,12 @@ We will see another example of this shortly.
 >
 > The last line of `index.md` includes the kind of information you might want to
-> re-use in multiple places throughout your site.
+> re-use in multiple places throughout your site as a footer.
```
```suggestion
> reuse in multiple places throughout your site as a footer.
```

to keep consistent with the changes made elsewhere in this PR.

tobyhodges commented on _episodes/includes.md:
```diff
@@ -237,53 +212,59 @@ We will see another example of this shortly.
-> then use an `include` statement to re-insert it
-> at the bottom of your site's `index.md`,
-> and add it at the bottom of the `about.md` pages.
+> Convert the above Markdown snippet into HTML and save it into `_includes/footer.html` file,
```
```suggestion
> Convert the above Markdown snippet into HTML and save it into a `_includes/footer.html` file,
```

tobyhodges commented on _episodes/includes.md:
```diff
@@ -237,53 +212,59 @@ We will see another example of this shortly.
-> at the bottom of your site's `index.md`,
-> and add it at the bottom of the `about.md` pages.
+> Convert the above Markdown snippet into HTML and save it into `_includes/footer.html` file,
+> then use the `include` directive to re-insert it
```
```
> then use the `include` directive to re-insert it
```

If we are using "reuse" now instead of "re-use," do you think this should be "reinsert?"

anenadic replied:
Sounds good - I have probably introduced a few more inconsistencies along the way - e.g. when we refer to "our repository" vs "your repository" interchangeably. The whole lesson would benefit from a language polish - will add a separate issue.

tobyhodges commented on _episodes/includes.md:
```diff
@@ -237,53 +212,59 @@ We will see another example of this shortly.
+> then use the `include` directive to re-insert it
+> at the bottom of your site's `index.md` page (making sure you remove the equivalent code). Also insert 
+> it at the bottom of the `about.md` page in a similar manner. We will explain 
+> why we meed the file to be in HTML rather than Markdown shortly.
```
```suggestion
> why we need the file to be in HTML rather than Markdown shortly.
```

tobyhodges commented on _episodes/layouts.md:
```diff
@@ -37,135 +37,99 @@ material often found at the very top or very bottom of a webpage.
+We will now define a new page layout called `default`,
+which we will save in a file `_layouts/default.html`.
+Having defined our navigational links in a separate file,
+we will start with a layout file that only includes this file:
```
```suggestion
we will start with a layout file that only includes this these links:
```

tobyhodges reviewed (changes requested):
This is a fantastic effort, @anenadic! A few suggestions, mostly minor typos and language polishing. We are almost good to go with this one 🎉

anenadic committed:
Update _episodes/includes.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/includes.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/includes.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/includes.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/includes.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/includes.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/layouts.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/includes.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/includes.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/includes.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/includes.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic committed:
Update _episodes/includes.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

anenadic commented:
Thanks for the review @tobyhodges ❤️ . I have committed all your changes (let me know if I've missed anything). I suggest I go now and resolve conflicts and merge the latest changes from gh-pages branch and then ask you to have another quick look and merge. Then the whole team can have look at the narrative and flow and fix any issues. Does that sound like a plan?

anenadic committed:
Added screenshots to illustrate header/footer. 'Reusable footer' -> 'Reusing' footer for consistency of headings.

anenadic committed:
Merged changes from gh-pages

anenadic commented:
Latest changes (8b475cdec190403bae91e421ccd9516f5afafe7e and 90bd4f78901bfefe77de24f488415a32495e554f): 
- Merged from gh-pages and resolved all conflicts.
- Added more screenshots to showcase adding of a header and a footer. 
- Restyled the footer section not to be in the form of an exercise but rather the main text (as there is quite a lot going on that may require the instructor to show). 
- Reverted the footer section name to 'Reusing Footer' instead of 'Reusable Footer' to be consistent with other headings in the episode. 
Review request pls @tobyhodges.

tobyhodges reviewed (approved):
Awesome, thank you @anenadic !
